common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java

```diff
@@ -343,8 +349,15 @@ public boolean hasNext() {
 
     @Override
     public ManagedBuffer next() {
-      final ManagedBuffer block = blockManager.getBlockData(
-        appId, execId, shuffleId, mapIds[mapIdx], reduceIds[mapIdx][reduceIdx]);
+      ManagedBuffer block;
+      if (!batchFetchEnabled) {
```

Comment:
nit: it's clearer to put different logic in different branches
```
if (!batchFetchEnabled) {
  ...
  if (reduceIdx < reduceIds[mapIdx].length - 1) {
    reduceIdx += 1;
  } else {
    reduceIdx = 0;
    mapIdx += 1;
  }
} else {
  assert reduceIds[mapIdx].length == 2
  block = ...reduceIds[mapIdx][0], reduceIds[mapIdx][1]);
  mapIdx += 1;
}
```

Reply:
Thanks, done in 3b6c5a6.